axelor-account/src/main/java/com/axelor/apps/account/web/InvoiceLineController.java

```diff
@@ -383,15 +395,182 @@ public void selectDefaultDistributionTemplate(ActionRequest request, ActionRespo
+          Beans.get(InvoiceLineAnalyticService.class);
+      invoiceLine = invoiceLineAnalyticService.selectDefaultDistributionTemplate(invoiceLine);
 
       response.setValue(
           "analyticDistributionTemplate", invoiceLine.getAnalyticDistributionTemplate());
       response.setValue(
           "analyticMoveLineList",
-          invoiceLineService.createAnalyticDistributionWithTemplate(invoiceLine));
+          invoiceLineAnalyticService.createAnalyticDistributionWithTemplate(invoiceLine));
+
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void createAnalyticAccountLines(ActionRequest request, ActionResponse response)
+      throws AxelorException {
+    try {
+      InvoiceLine invoiceLine = request.getContext().asType(InvoiceLine.class);
+      if (request.getContext().getParentContext() != null) {
+        Invoice invoice = request.getContext().getParent().asType(Invoice.class);
+        invoiceLine =
+            Beans.get(InvoiceLineAnalyticService.class).analyzeInvoiceLine(invoiceLine, invoice);
+        response.setValue("analyticMoveLineList", invoiceLine.getAnalyticMoveLineList());
+      }
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void setAxisDomains(ActionRequest request, ActionResponse response)
+      throws AxelorException {
+    try {
+      InvoiceLine invoiceLine = request.getContext().asType(InvoiceLine.class);
+      if (request.getContext().getParentContext() != null) {
+        Invoice invoice = request.getContext().getParent().asType(Invoice.class);
+        List<Long> analyticAccountList = new ArrayList<Long>();
+
+        for (int i = startAxisPosition; i <= endAxisPosition; i++) {
+          if (invoice != null
+              && Beans.get(AnalyticToolService.class)
+                  .isPositionUnderAnalyticAxisSelect(invoice.getCompany(), i)) {
+            analyticAccountList =
+                Beans.get(InvoiceLineAnalyticService.class).getAxisDomains(invoiceLine, invoice, i);
+            if (CollectionUtils.isEmpty(analyticAccountList)) {
+              response.setAttr(
+                  "axis".concat(Integer.toString(i)).concat("AnalyticAccount"),
+                  "domain",
+                  "self.id IN (0)");
+            } else {
+              String idList =
+                  analyticAccountList.stream()
+                      .map(id -> id.toString())
+                      .collect(Collectors.joining(","));
+              response.setAttr(
+                  "axis".concat(Integer.toString(i)).concat("AnalyticAccount"),
+                  "domain",
+                  "self.id IN (" + idList + ")");
+            }
+          }
+        }
+      }
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void setRequiredAnalyticAccount(ActionRequest request, ActionResponse response)
+      throws AxelorException {
```

Comment:
Please remove throw declaration when the method in fact does not throw the exception